scripts/run_presubmit_checks.sh

```diff
@@ -47,11 +61,47 @@ python $(dirname $0)/pre_commit_linter.py || exit 1
 echo 'Linting passed.'
 echo ''
 
-# Run frontend unit tests.
-echo 'Running frontend unit tests'
-source $(dirname $0)/run_frontend_tests.sh --run-minified-tests=true || exit 1
-echo 'Frontend tests passed.'
-echo ''
+# Read arguments from the command line.
+for i in "$@"
+do
+case $i in
+    -b=*|--branch=*)
+    ORIGIN_BRANCH=${i#*=}
+    shift
+    ;;
+esac
+done
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+# If the current branch exists on remote origin, MATCHED_BRANCH_NUM=1
+# else MATCHED_BRANCH_NUM=0
+MATCHED_BRANCH_NUM=$(git ls-remote --heads origin $CURRENT_BRANCH | wc -l)
+# Set the origin branch to develop if it's not specified.
+if [ -n "$ORIGIN_BRANCH" ]; then
+  BRANCH=$ORIGIN_BRANCH
+elif [ $MATCHED_BRANCH_NUM == 1 ]; then
+  BRANCH=origin/$CURRENT_BRANCH
+else
+  BRANCH=develop
+fi
+
+FRONTEND_DIR='core/templates/dev/head'
+
+echo "Comparing the current branch with $BRANCH"
+
+if [ -n "$(git diff --cached --name-only --diff-filter=ACM ${BRANCH} | grep ${FRONTEND_DIR})" ]
```

Comment:
Ah, take a look at this maybe: https://superuser.com/a/750913

Gives you an explanation of why there are no brackets as well as introducing the -q arg. If you decide to go this route, might be worth referring to this link in a comment.

Apart from that, I think this PR looks good. Once you confirm (by a top-level comment on this PR) that you have manually tested it, it can go in. The manual testing is necessary, I think; there are often hidden issues in bash scripts, and neither you nor I are super familiar with the language, so we need to make sure that this works correctly end-to-end in all cases.

Also, in your comment, please provide a specific list of which scenarios you have manually tested, for future reference -- that way, if this breaks for someone, we can see if it wasn't included here in the first place, or if there's some kind of system-dependent behaviour. Thanks!